server/src/main/java/com/linecorp/centraldogma/server/internal/api/MirroringServiceV1.java

```diff
@@ -0,0 +1,152 @@
+import com.linecorp.centraldogma.server.metadata.ProjectRole;
+import com.linecorp.centraldogma.server.mirror.Mirror;
+import com.linecorp.centraldogma.server.storage.project.ProjectManager;
+
+/**
+ * Annotated service object for managing mirroring service.
+ */
+@ProducesJson
+@RequiresRole(roles = ProjectRole.OWNER)
+@ExceptionHandler(HttpApiExceptionHandler.class)
+public class MirroringServiceV1 extends AbstractService {
+
+    private final MetadataService mds;
+
+    public MirroringServiceV1(ProjectManager projectManager, CommandExecutor executor, MetadataService mds) {
+        super(projectManager, executor);
+        this.mds = mds;
+    }
+
+    /**
+     * GET /projects/{projectName}/mirrors
+     *
+     * <p>Returns the list of the mirrors in the project.
+     */
+    @Get("/projects/{projectName}/mirrors")
+    public CompletableFuture<List<MirrorDto>> listMirrors(@Param String projectName) {
+        return projectManager().get(projectName).metaRepo().mirrors(true).thenApply(mirrors -> {
+            return mirrors.stream()
+                          .map(mirror -> convertToMirrorDto(projectName, mirror))
+                          .collect(toImmutableList());
+        });
+    }
+
+    /**
+     * POST /projects/{projectName}/mirrors
+     *
+     * <p>Creates a new mirror.
+     */
+    @Post("/projects/{projectName}/mirrors")
+    @ConsumesJson
+    @StatusCode(201)
+    public CompletableFuture<Revision> createMirror(@Param String projectName, MirrorDto newMirror,
+                                                    Author author) {
+        validateMirror(newMirror);
+        return mds.createMirror(projectName, newMirror, author);
+    }
+
+    private static void validateMirror(MirrorDto mirror) {
+        checkArgument(!Strings.isNullOrEmpty(mirror.id()), "Mirror ID is empty");
+        final Cron schedule = MirrorConfig.cronParser.parse(mirror.schedule());
+        final CronField secondField = schedule.retrieve(CronFieldName.SECOND);
+        checkArgument(!secondField.getExpression().asString().contains("*"),
+                      "The second field of the schedule must be specified. (seconds: *, expected: 0-59)");
+    }
+
+    /**
+     * GET /projects/{projectName}/mirrors/{index}
+     *
+     * <p>Returns the mirror at the specified index in the project mirror list.
+     */
+    @Get("/projects/{projectName}/mirrors/{index}")
+    public CompletableFuture<MirrorDto> getMirror(@Param String projectName, @Param int index) {
+        checkArgument(index >= 0, "index: %s (expected: >= 0)", index);
+
+        return projectManager().get(projectName).metaRepo().mirrors(true).thenApply(mirrors -> {
+            if (index >= mirrors.size()) {
+                throw new EntryNotFoundException(
+                        "No such mirror at the index " + index + " in " + projectName);
+            }
+            return convertToMirrorDto(projectName, mirrors.get(index));
```

Comment:
It seems like we're just using the index and not actually comparing `mirrors.index` with `index` from the parameters.
What do you think of just removing the index from the DTO altogether for simplicity if we know it's not going to change?

I didn't think there is any sorting from any point (since this is an array of json objects, there probably isn't a natural sorting order anyways)